lib/ledenet/api.rb

```diff
@@ -14,56 +14,112 @@ def initialize(device_address, options = {})
 
     def on
       send_bytes_action(0x71, 0x23, 0x0F, 0xA3)
-
       true
     end
 
     def off
       send_bytes_action(0x71, 0x24 ,0x0F, 0xA4)
-
       true
     end
 
     def on?
-      (status.bytes[13] & 0x01) == 1
+      # Bulbs respond with "35" or "36" (HEX: 23 or 24)
+      # for on and off respectively
+      power_state = status[2].unpack('C').to_s.delete('[]')
+      if  power_state == "35"
+        true
+      else
+        false
+      end
     end
 
-    def update_color(r, g, b)
-      checksum = color_checksum(r, g, b)
+    def update_color(r,g,b) # Supports legacy color Updates
+      update_ufo(r,g,b,0,false)
+      true
+    end
 
-      send_bytes_action(0x31, r, g, b, 0xFF, 0, 0x0F, checksum)
+    def update_ufo(r,g,b,w,persist) # Update a UFO wireless device
+      msg = Array.new
+      if persist
```

Comment:
Seems like you could pull this logic into a method, especially since it's mostly repeated in `update_bulb_color` and `update_bulb_white`.